TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/activities/RunningActivity.java

```diff
@@ -364,6 +387,40 @@ public void onClick(View view) {
         }
     }
 
+    public void onClickOkLoadOccupancyGridDialog(final String occupancyGridName) {
+        if (occupancyGridName == null || occupancyGridName.isEmpty()) {
+            Log.e(TAG, "Name is null or empty, unable to load the occupancy grid");
+            displayToastMessage(R.string.name_error);
```

Comment:
The functionality to display the dialog again was added in a previous commit. 
Also, the OK button is now disabled if the text field is empty, so you are right it should never happen. I will add an assert here.